primitives/runtime/src/generic/checked_extrinsic.rs

```diff
@@ -50,14 +50,17 @@ where
 
 	fn validate<U: ValidateUnsigned<Call = Self::Call>>(
 		&self,
+		// TODO [ToDr] should source be passed to `SignedExtension`s?
```

Comment:
I think answering this highly depends on the ongoing _existential crisis_ around `ValidateUnsigned` in https://github.com/paritytech/substrate/pull/5006.

Reply:
I think ideally ValidateUnsignedShould be gone (I think a fix to the main issue that Basti has there would be doable, there must be some default guard in place to reject everything except for when a module explicitly says otherwise) and then Source need to be passed to SignedExtension.

Reply:
Yup, that's why I left a TODO for now, I feel it's going to be addressed in #5006 indeed, just depends on whichever PR goes in first :)
Added the PR number to the note.